streams/src/main/java/org/apache/kafka/streams/kstream/internals/InternalStreamsBuilder.java

```diff
@@ -78,7 +78,8 @@ public InternalStreamsBuilder(final InternalTopologyBuilder internalTopologyBuil
 
     public <K, V> KStream<K, V> stream(final Collection<String> topics,
                                        final ConsumedInternal<K, V> consumed) {
-        final String name = newProcessorName(KStreamImpl.SOURCE_NAME);
+
+        final String name = new NamedInternal(consumed.name()).orElseGenerateWithPrefix(this, KStreamImpl.SOURCE_NAME);
```

Comment:
I am personally a little confused what `orElseGenerateWithPrefix` means? It's a personal preference, but I don't think it's easy to read. Similar for `suffixWithOrElseGet`. (Maybe it's just me, being not use to fancy Java8 constructs that are mimicked here...)

Curious to hear what others think.

Reply:
I'm ok with the names, but I don't have a strong opinion.  We still have time to address between now and the final PR though.